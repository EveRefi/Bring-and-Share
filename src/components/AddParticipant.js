import React, { useState } from "react";
import axios from "axios";
import "../styles/AddParticipant.css";

const AddParticipant = ({
  eventId,
  fetchParticipants,
  setFetchParticipants,
}) => {
  const initialState = {
    fields: {
      name: "",
      email: "",
      toBring: "",
      dislikes: "",
      dietInfo: "",
      eventID: eventId,
    },
  };

  const [fields, setFields] = useState(initialState.fields);
  const [addParticipantsButton, setAddParticipantsButton] = useState(false);
  const showaddparticipants = () => {
    setAddParticipantsButton(true);
  };
  const closeWindow = () => {
    setAddParticipantsButton(false);
  };
  const handleAddPerson = (e) => {
    e.preventDefault();

    axios
      .post(
        `https://final-mcrcodes-project.herokuapp.com/events/${eventId}/participants`,
        fields
      )
      .then(() => {
        setFetchParticipants(!fetchParticipants);
      })
      .catch((error) => {
        console.log(error);
      });
  };
  const handleFieldChange = (e) => {
    setFields({ ...fields, [e.target.name]: e.target.value });
  };

  return (
    <>
      <div className="theContainerforAddP">
        <div className="buttonContainer">
          <button
            data-testid="addParticipantNavButton"
            className="addParticipantNavButton"
            onClick={() => showaddparticipants()}
          >
            Press to Add Attendees
          </button>
        </div>
        {addParticipantsButton === true && (
          <div className="addParticipant">
            <form
              data-testid="addParticipantForm"
              className="addParticipantForm"
              onSubmit={handleAddPerson}
            >
              <p className="closeX">
                <button
                  className="closeAddParticipantButton"
                  onClick={closeWindow}
                >
                  x
                </button>
              </p>
              <h3 className="addParticipantTitle">Add your participants</h3>
              <div class="addParticipantFormContainer">
                <div className="name">
                  <label htmlFor="name">
                    Name
                    <input
                      type="text"
                      id="name"
                      name="name"
                      data-testid="name"
                      value={fields.name}
                      onChange={handleFieldChange}
                      placeholder="Your name"
                    />
                  </label>
                </div>

                <div className="email">
                  <label htmlFor="email">
                    Email <br />
                    <input
                      type="text"
                      id="email"
                      name="email"
                      data-testid="email"
                      value={fields.email}
                      onChange={handleFieldChange}
                      placeholder="youremail@email.com"
                    />
                  </label>
                </div>

                <div className="toBring">
                  <label htmlFor="toBring">
                    I am bringing... <br />
                    <input
                      type="text"
                      id="toBring"
                      name="toBring"
                      data-testid="toBring"
                      value={fields.toBring}
                      onChange={handleFieldChange}
                      placeholder="Ferrero Rocher"
                    />
                  </label>
                </div>

                <div className="dislikes">
                  <label htmlFor="dislikes">
                    Food dislikes <br />
                    <input
                      type="text"
                      id="dislikes"
                      name="dislikes"
                      data-testid="dislikes"
                      value={fields.dislikes}
                      onChange={handleFieldChange}
                      placeholder="I don't like..."
                    />
                  </label>
                </div>

                <div className="dietInfo">
                  <label htmlFor="dietInfo">
                    Diet info <br />
                    <p className="leaveBlankP">
                      Leave blank if you have no diet requirements
                    </p>
                    <input
                      type="text"
                      id="dietInfo"
                      name="dietInfo"
                      data-testid="dietInfo"
                      value={fields.dietInfo}
                      onChange={handleFieldChange}
                      placeholder="Any food restrictions?"
                    />
                  </label>
                </div>

                <div className="addParticipantButton">
                  <button
                    data-testid="addButton"
                    className="addButton"
                    type="submit"
                  >
                    Add
                  </button>
                </div>
              </div>
            </form>
          </div>
        )}
      </div>
    </>
  );
};

export default AddParticipant;
