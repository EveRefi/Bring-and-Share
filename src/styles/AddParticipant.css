@import url("https://fonts.googleapis.com/css2?family=Roboto&display=swap");

.addParticipant {
  background-color: papayawhip;
  border: 5px solid purple;
  font-family: "Roboto", sans-serif;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  width: 250px;
}

.name,
.toBring,
.dislikes,
.dietInfo,
.addButton {
  width: 200px;
  padding: 10px;
  margin-left: auto;
  margin-right: auto;
  border: 5px solid magenta;
}

.addButton {
  padding: 10px;
  border-radius: 10px;
  color: rgb(4, 4, 4);
  font-weight: 700;
  background-color: rgb(250, 237, 58);
  border: none;
  box-shadow: 5px 5px #888888;
}
