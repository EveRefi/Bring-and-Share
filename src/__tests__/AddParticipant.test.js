import React from "react";
import { render } from "@testing-library/react";
import AddParticipant from "../components/AddParticipant";


describe('AddParticipant component', () => {

    const mockId = 123

    it('renders correctly', () => {

        const {asFragment} = render (<AddParticipant eventId={mockId}/>)

        expect(asFragment()).toMatchSnapshot()
    })

    it('renders the correct form elements', () => {
        const { getByTestId } = render (<AddParticipant eventId={mockId}/>)

        expect(getByTestId('name')).toBeValid()
        expect(getByTestId('toBring')).toBeValid()
        expect(getByTestId('dislikes')).toBeValid()
        expect(getByTestId('dietInfo')).toBeValid()
        expect(getByTestId('addButton')).toBeInTheDocument()

    })
})