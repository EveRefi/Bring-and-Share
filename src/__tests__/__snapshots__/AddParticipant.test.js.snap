// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddParticipant component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="addParticipant"
  >
    <h2>
      Add your participants
    </h2>
    <p>
      Fill in all fields and click "create event"
    </p>
    <form>
      <div
        class="name"
      >
        <label
          for="name"
        >
          Name 
          <br />
          <input
            data-testid="name"
            id="name"
            name="name"
            placeholder="Name of fellow Party Queen..."
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="toBring"
      >
        <label
          for="toBring"
        >
          I am bringing... 
          <br />
          <input
            data-testid="toBring"
            id="toBring"
            name="toBring"
            placeholder="Ferrero Rocher"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="dislikes"
      >
        <label
          for="dislikes"
        >
          I don't like... 
          <br />
          <input
            data-testid="dislikes"
            id="dislikes"
            name="dislikes"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="dietInfo"
      >
        <label
          for="dietInfo"
        >
          Diet info 
          <br />
          <input
            data-testid="dietInfo"
            id="dietInfo"
            name="dietInfo"
            placeholder="Vegetarian? Kosher? Coeliac?"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="createEventButton"
      >
        <button
          class="addButton"
          data-testid="addButton"
          type="submit"
        >
          Add
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
